hotsos/core/ycheck/events.py

```diff
@@ -235,123 +480,115 @@ class EventHandlerBase(YHandlerBase, EventProcessingUtils):
                       event.input.command is not None)
 
             section_name = event.parent.name
-            if section_name not in _event_defs:
-                _event_defs[section_name] = {}
-
-            for path in event.input.paths:
-                if event.input.command:
-                    # don't apply constraints to command outputs
-                    allow_constraints = False
-                else:
-                    allow_constraints = True
-
-                event.search.load_searcher(self.searcher, path,
-                                           allow_constraints=allow_constraints)
+            if section_name not in _events:
+                _events[section_name] = {}
 
-            passthrough = bool(event.search.passthrough_results)
-            emeta = {'passthrough': passthrough,
-                     'sequence': event.search.sequence_search,
-                     'tag': event.search.unique_search_tag}
-            _event_defs[section_name][event.name] = emeta
+            _events[section_name][event.name] = event.resolve_path
 
-        return _event_defs
-
-    def load(self):
-        """ Pre-load event definitions. """
-        self.event_definitions
+        return _events
 
     @property
     def final_event_results(self):
         """ Cache of results in case run() is called again. """
         return self._event_results
 
-    def run(self, results):
+    def run(self, results=None):
         """
         Process each event and call respective callback functions when results
         where found.
 
-        @param results: SearchResultsCollection object.
+        @param results: If no results are provides we get them from the global
+                        searcher. This is provided for the case where a custom
+                        searcher is in use.
         """
+        if results is None:
+            results = self.search_registry.get_global_searcher_results()
+
         if self.final_event_results is not None:
             return self.final_event_results
 
         if not CALLBACKS:
             raise Exception("need to register at least one callback for "
                             "event handler.")
 
-        log.debug("registered callbacks:\n%s", '\n'.join(CALLBACKS.keys()))
+        log.debug("registered event callbacks:\n%s", '\n'.
+                  join(CALLBACKS.keys()))
         info = {}
-        for section_name, section in self.event_definitions.items():
-            for event, event_meta in section.items():
-                search_tag = event_meta['tag']
+        for section_name, section in self.events.items():
+            for event, fullname in section.items():
+                event_search = self.search_registry[fullname]['search']
+                search_tag = event_search.unique_search_tag
                 seq_def = None
-                if event_meta['passthrough']:
+                if bool(event_search.passthrough_results):
```

Comment:
I think instead of casting this to `bool` here we should ensure that the property always returns a bool, i.e.:

```python
    @property
    def passthrough_results(self):
        if not isinstance(self.content, dict):
            return False

        return bool(self.content.get('passthrough-results', False))
```

Reply:
agreed